agora/Agora/Proposal/Scripts.hs

```diff
@@ -533,44 +532,45 @@ proposalValidator proposal =
 
           ----------------------------------------------------------------------
 
-          PUnlock r -> withSingleStake $ \stakeInF stakeOut _ -> do
+          PUnlock _ -> withSingleStake $ \stakeInF stakeOut _ -> do
             -- At draft stage, the votes should be empty.
             pguardC "Shouldn't retract votes from a draft proposal" $
               pnot #$ currentStatus #== pconstant Draft
```

Comment:
Arguably, this check should be redundant. If the invariant holds that it's impossible to create votes at a draft proposal, then the other checks will ensure that it's impossible to retract them, too.

Reply:
In some edge cases, it does help a ittle bit in terms of performance. It makes perfect senses to just remove it though.